### PR TITLE
Fix config editor text entry focus

### LINKED ISSUE #804: Config editor text fields lose focus on letter press

### PR DESCRIPTION
**Pull request by garrettjstevens**
This fixes #804 and shouldn't regress on #772

### CODE REVIEW

**Comment by garrettjstevens** [comment]
Wait, this is still broken. Let me try again.

**Comment by garrettjstevens** [comment]
Ok, now I think it's actually working. Re-opening.

File: packages/config/src/ConfigurationEditorDrawerWidget/components/ConfigurationEditor.js [hunk 1]
```diff
@@ -94,8 +95,9 @@ const useStyles = makeStyles(theme => ({
 
 function ConfigurationEditor({ model }) {
   const classes = useStyles()
+  const key = model.target && readConfObject(model.target, 'trackId')
   return (
-    <div className={classes.root} data-testid="configEditor">
+    <div className={classes.root} key={key} data-testid="configEditor">
```
Comment (cmdcolin) [inline packages/config/src/ConfigurationEditorDrawerWidget/components/ConfigurationEditor.js]: why does this entry need a key

Comment (cmdcolin) [reply packages/config/src/ConfigurationEditorDrawerWidget/components/ConfigurationEditor.js]: or i guess similarly, why is the key removed below and added here

Comment (garrettjstevens) [reply packages/config/src/ConfigurationEditorDrawerWidget/components/ConfigurationEditor.js]: When switching over from one config editor drawer widget to another that look very similar (e.g. editors for NCList and BigBed tracks), React sometimes thinks they are actually the same thing and just doesn't update (to save rerenders). That's what caused #772, it wasn't updating the callback editor between configs and so the second track ended up with the first's callback. It needs a key to tell it that two things are different. I put the key on the config slot itself before, but that caused it to rerender every time something in the slot changed, which is why it lost focus. I now realize, though, that the parent can have a different key and it updates just as well, so I put the key there instead, and since the key doesn't change every time one of the slots change, it doesn't lose focus.

Comment (cmdcolin) [reply packages/config/src/ConfigurationEditorDrawerWidget/components/ConfigurationEditor.js]: Ok, thanks. Can you add a comment where you put this line of code in to indicate something to this effect

Comment (garrettjstevens) [reply packages/config/src/ConfigurationEditorDrawerWidget/components/ConfigurationEditor.js]: ok, done
